enhancements/cloud-integration/azure/azure-workload-identity.md

```diff
@@ -0,0 +1,358 @@
+	if strings.TrimSpace(config.ClientID) == "" {
+		config.ClientID = os.Getenv("AZURE_CLIENT_ID")
+		if strings.TrimSpace(config.ClientID) == "" {
+			return nil, errors.New("empty client ID")
+		}
+	}
+
+	// Fallback to using client secret from env variable if not set.
+	if strings.TrimSpace(config.ClientSecret) == "" {
+		config.ClientID = os.Getenv("AZURE_CLIENT_SECRET")
+		// Skip validation; fallback to token (below) if env variable is also not set.
+	}
+
+	var (
+		cred azcore.TokenCredential
+		err  error
+	)
+	if strings.TrimSpace(config.ClientSecret) == "" {
+		options := azidentity.WorkloadIdentityCredentialOptions{
+			ClientOptions: azcore.ClientOptions{
+				Cloud: cloudConfig,
+			},
+		}
+		cred, err = azidentity.NewWorkloadIdentityCredential(config.TenantID, config.ClientID, "/var/run/secrets/openshift/serviceaccount/token", &options)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		options := azidentity.ClientSecretCredentialOptions{
+			ClientOptions: azcore.ClientOptions{
+				Cloud: cloudConfig,
+			},
+		}
+		cred, err = azidentity.NewClientSecretCredential(config.TenantID, config.ClientID, config.ClientSecret, &options)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	scope := endpointToScope(config.Environment.TokenAudience)
+
+	// Use an adapter so azidentity in the Azure SDK can be used as
+	// Authorizer when calling the Azure Management Packages, which we
+	// currently use. Once the Azure SDK clients (found in /sdk) move to
+	// stable, we can update our clients and they will be able to use the
+	// creds directly without the authorizer. The schedule is here:
+	// https://azure.github.io/azure-sdk/releases/latest/index.html#go
+	authorizer := azidext.NewTokenCredentialAdapter(cred, []string{scope})
+
+	return authorizer, nil
+}
+```
+
+#### Mutating admission webhook
+
+CCO will deploy and lifecycle the [Azure Workload Identity mutating admission webhook](https://azure.github.io/azure-workload-identity/docs/installation/mutating-admission-webhook.html) on Azure clusters such that users can annotate workload `ServiceAccounts` with Managed Identity details necessary for creating clients. When the mutating admission webhook finds these annotations on a
+`ServiceAccount` referenced by a pod being created, environment variables are set for the pod for the `AZURE_CLIENT_ID`, `AZURE_TENANT_ID` and `AZURE_FEDERATED_TOKEN_FILE`.
+
+This will be similar to how CCO deploys the [AWS Pod Identity webhook](https://github.com/openshift/aws-pod-identity-webhook) which we have forked for use by user workloads.
+
+#### Variation [optional]
+
+TBD
+
+### API Extensions
+
+None as of now.
+
+### Implementation Details/Notes/Constraints [optional]
+
+TBD
+
+### Risks and Mitigations
+
+- The feature this work relies on was recently made public preview. What is the timeline for GA for Workload identity federation support for Managed Identities?
+- How will security be reviewed and by whom?
+- How will UX be reviewed and by whom?
```

Comment:
If we match the azure recommendation, I'm ok with pain it induces.  If we change the azure t he documentation/recommendation, I think this (like the security point above), deserve close scrutiny.